dashboards-observability/.cypress/integration/event_analytics.spec.js

```diff
@@ -615,4 +615,124 @@ describe('Renders data view', () => {
     cy.get('[data-test-subj="workspace__dataTableViewSwitch"]').click();
     cy.get('[data-test-subj="workspace__dataTable"]').should('not.exist');
   });
-});
\ No newline at end of file
+});
+
+describe('Data grid control section/Table view validation', () => {
+  beforeEach(() => {
+    landOnEventVisualizations();
+    querySearch(TEST_QUERIES[5].query, TEST_QUERIES[5].dateRangeDOM);
+    cy.get('[data-test-subj = "workspace__dataTableViewSwitch"]').click();
+    cy.wait(delay);
+  });
+
+  it('Search columns', () => {
+    cy.get('[data-test-subj = "dataGridColumnSelectorButton"]').click();
+    cy.get('[aria-label = "Search columns"]').click().type('destcountry');
+    cy.get('.euiSwitch__label').eq(1).contains('DestCountry').should('exist');
+  });
+
+  it('Toggle columns and verify the columns hidden text', () => {
+    cy.get('[data-test-subj = "dataGridColumnSelectorButton"]').click();
+    cy.get('.euiSwitch.euiSwitch--compressed.euiSwitch--mini .euiSwitch__button').eq(3).click();
+    cy.get('.euiButtonEmpty__text').eq(3).click().should('have.text', '1 column hidden');
+  });
+
+  it('Show all button', () => {
+    cy.get('[data-test-subj = "dataGridColumnSelectorButton"]').click();
+    cy.get('.euiPopoverFooter .euiFlexItem.euiFlexItem--flexGrowZero').eq(0).should('have.text', 'Show all').click();
+    cy.get('.euiDataGrid__focusWrap').click().should('exist');
+  });
+
+  it('Hide all button', () => {
+    cy.get('[data-test-subj = "dataGridColumnSelectorButton"]').click();
+    cy.get('.euiPopoverFooter .euiFlexItem.euiFlexItem--flexGrowZero').eq(1).should('have.text', 'Hide all').click();
+    cy.get('.euiDataGrid__focusWrap').click().should('exist');
+    cy.get('[data-test-subj="dataGridColumnSelectorPopover"]').should('have.text', '4 columns hidden');
+  });
+
+  it('Render visualization for table view and see data in Full screen', () => {
+    cy.get('.euiDataGridHeaderCell__content').contains('max(AvgTicketPrice)').should('exist');
```

Comment:
Instead of writing `cy.get('.euiDataGridHeaderCell__content')`, can't we just store that in a variable and refer it every time? `cy.get` actually searches the dom and get the required node every time. I think, We can improve on the execution time of the test cases with this.

Reply:
created function for repetitive code.